### PR TITLE
ActiveStorage support

### PR DESCRIPTION
**Pull request by bishosilwal**
This pr has the following changes:
 - [x] Add global namespace `ImageKitIo`
 - [x] Refactor and multiple changes
 - [x] Fix failed tests
 - [x] Add ActiveStorage support
 - [x] Implement api changes
 - [x] Update README.md
 - [x] Delete application inside samples folder in this repo.

Related work
 - [ ] Update sample [quickstart project for ruby](https://github.com/imagekit-samples/quickstart/tree/master/ruby-on-rails) to showcase basic integration of all API operation along with Carrierwave and activestorage.
 - [ ] Update [getting started guide](https://docs.imagekit.io/getting-started/quickstart-guides/ruby-on-rails) by submitting a PR against `dev` branch of [this repo](https://github.com/imagekitio/docs-keep-a-changelog).

### CODE REVIEW

File: test/imagekit/utils/formatter_test.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  let(:subject) { described_class }
+
+  it "snake_to_camel_test" do
+    result=subject.snake_to_camel('my_snake_to_camel')
+    expect('mySnakeToCamel').to eq(result)
+  end
+
+  it "camel_to_snake_test" do
+    result=subject.camel_to_snake('mySnakeToCamel')
+    expect('my_snake_to_camel').to eq(result)
```
Comment (imagekitio) [inline test/imagekit/utils/formatter_test.rb:14]: Although technically same thing, but shouldn't it be written like this
```ruby
expect(result).to eq('my_snake_to_camel')
```

Comment (bishosilwal) [reply test/imagekit/utils/formatter_test.rb:14]: Yeah, It should be the way you said but still, it works. The code was written by the previous developer, I didn't change anything besides the namespace here.

File: lib/active_storage/service/ik_file.rb [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      def thumbnail_url
+        identifier['thumbnailUrl']
+      end
+
+      def filename
+        identifier['name']
+      end
+
+      def size
+        identifier['size']
```
Comment (imagekitio) [inline lib/active_storage/service/ik_file.rb:36]: There are more parameters in the file response, please check API reference - https://docs.imagekit.io/api-reference/upload-file-api/server-side-file-upload#understanding-response

**Review by imagekitio — CHANGES_REQUESTED** [review]

File: README.md [hunk 3]
```diff
@@ -376,37 +418,173 @@ imagekitio.update_file_details(file_id, {
...
+```
+
+**20. Create Custom Metadata Fields**
+
+Create custom metadata fields as per the [API documentation here](https://docs.imagekit.io/api-reference/custom-metadata-fields-api/create-custom-metadata-field)
+
+```ruby
+
+imagekitio.create_custom_metadata_fields(
+    'price',
```
Comment (imagekitio) [inline README.md:553]: Let's keep the whole request as JSON, similar to the API.

File: README.md [hunk 4]
```diff
@@ -376,37 +418,173 @@ imagekitio.update_file_details(file_id, {
...
+```ruby
+imagekitio.get_custom_metadata_fields
+```
+
+**22. Update Custom Metadata Fields**
+
+Update custom metadata fields as per the [API documentation here](https://docs.imagekit.io/api-reference/custom-metadata-fields-api/update-custom-metadata-field)
+
+```ruby
+imagekitio.update_custom_metadata_fields('file_id', label: 'custom-price', schema: nil)
```
Comment (imagekitio) [inline README.md:576]: `file_Id` seems to be incorrect. Let's keep the whole request as JSON, similar to the API.

File: README.md [hunk 5]
```diff
@@ -376,37 +418,173 @@ imagekitio.update_file_details(file_id, {
...
+```
+
+**23. Delete Custom Metadata Fields**
+
+Delete custom metadata fields as per the [API documentation here](https://docs.imagekit.io/api-reference/custom-metadata-fields-api/delete-custom-metadata-field)
+
+```ruby
+imagekitio.delete_custom_metadata_fields('file_id')
```
Comment (imagekitio) [inline README.md:584]: `file_id` seems to be incorrect. Let's keep the whole request as JSON, similar to the API.

File: README.md [hunk 6]
```diff
@@ -469,51 +647,21 @@ imagekitio.phash_distance('a4a65595ac94518b', '7838873e791f8400')
 
 ## Sample Application
 There are two sample apps:
```
Comment (imagekitio) [inline README.md:649]: Three, not two. Please proofread the readme once.

**Review by imagekitio — CHANGES_REQUESTED** [review]
Add examples of how to send `extensions` and `customMetadata` in upload and update file operations..

**Review by imagekitio — CHANGES_REQUESTED** [review]
Test cases are not complete. For E.g. where is the test case that verifies that the user can send `extension` and `customMetadata` in upload and update API and SDK will stringify it before hitting API.

**Comment by imagekitio** [comment]
In the list of supported transformation parameters add `raw`. 

Basically, if the user wants to generate transformations in their application and add them to the URL as it is, they can use the raw parameter.

Accordingly update the test case with name `test_generate_url_should_have_transformation_parameters`

**Review by imagekitio — CHANGES_REQUESTED** [review]
In the list of supported transformation parameters add `raw`. 

Basically, if the user wants to generate transformations in their application and add them to the URL as it is, they can use the raw parameter.

Accordingly update the test case with name `test_generate_url_should_have_transformation_parameters`

**Pull request merged**